### PR TITLE
improve deadLetters and logging when send queue overflow

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala [hunk 1]
```diff
@@ -89,18 +89,20 @@ private[akka] object RemoteActorRefProvider {
     import EndpointManager.Send
 
     override def !(message: Any)(implicit sender: ActorRef): Unit = message match {
-      case Send(m, senderOption, _, seqOpt) ⇒
+      case Send(m, senderOption, recipient, seqOpt) ⇒
         // else ignore: it is a reliably delivered message that might be retried later, and it has not yet deserved
         // the dead letter status
-        if (seqOpt.isEmpty) super.!(m)(senderOption.orNull)
+        if (seqOpt.isEmpty) super.!(DeadLetter(m, senderOption.getOrElse(_provider.deadLetters), recipient))
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala]: Then the comment above should go also?

**Comment by johanandren** [comment]
LGTM

File: akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala (hunk 1 above)
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala]: I don't understand. the comment `else ignore` refers to `if (seqOpt.isEmpty)`. I hope I didn't change anything of that.

Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala]: The reason I changed this is that deadletters from remoting didn't contain the `recipient`, which can be very useful when debugging.

Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala]: Ah, mentally skipped the else part, sorry, you are right!

**Comment by ktoso** [comment]
LGTM

**Comment by patriknw** [comment]
@ktoso Thanks for merging, but I wanted to try the new github feature that you can re-target a PR. I'll do that with the others.

**Comment by ktoso** [comment]
Ouch, I'm sorry for jumping ahead then :<

**Pull request merged**